src/ExpressionEvaluator/Core/Source/ExpressionCompiler/DkmUtilities.cs

```diff
@@ -26,7 +26,7 @@ private static IEnumerable<DkmClrModuleInstance> GetModulesInAppDomain(this DkmC
             var appDomainId = appDomain.Id;
             return runtime.GetModuleInstances().
                 Cast<DkmClrModuleInstance>().
-                Where(module => module.AppDomain.Id == appDomainId);
+                Where(module => !module.AppDomain.IsUnloaded && module.AppDomain.Id == appDomainId);
```

Comment:
What if the AppDomain is unloaded after this point but before the caller uses the module? If there's no such race condition, then this method could check appDomain.IsUnloaded up front rather than in the Where clause.